samples/drivers/espi/src/main.c

```diff
@@ -0,0 +1,339 @@
+		loop_cnt--;
+	} while (loop_cnt > 0);
+	if (loop_cnt == 0) {
+		printk("PWRT! for %x %x\n", pin, level);
+		return -ETIMEDOUT;
+	}
+	return 0;
+}
+
+int wait_for_vwire(struct device *espi_dev, enum espi_vwire_signal signal,
+			u16_t timeout, u8_t exp_level)
+{
+	int ret;
+	u8_t level;
+	u16_t loop_cnt = timeout;
+
+	do {
+		ret = espi_receive_vwire(espi_dev, signal, &level);
+		if (ret) {
+			printk("Failed to read %x %d", signal, ret);
+			return -EIO;
+		}
+		if (exp_level == level) {
+			break;
+		}
+		k_busy_wait(50);
+		loop_cnt--;
+	} while (loop_cnt > 0);
+	if (loop_cnt == 0) {
+		printk("VWIRE %d! is %x\n", signal, level);
+		return -ETIMEDOUT;
+	}
+	return 0;
+}
+
+int pwr_init(void)
+{
+	int ret = 0;
+
+	printk("eSPI test - Power initialization...");
+
+	ret = gpio_pin_write(gpio_devC, BOARD_RUN_PIN, 1);
+	if (ret)
+		printk("Failed to write %x %d", BOARD_RUN_PIN, ret);
+
+	ret = wait_for_pin(gpio_devA, 1000, BOARD_PWRGD_PIN, 1);
+	if (ret) {
+		printk("Timeout!");
+		return ret;
+	}
+
+	k_sleep(100);
+
+	ret = gpio_pin_write(gpio_devB, ESPI_MASTER_INIT_PIN, 1);
+	if (ret)
+		printk("Failed to write %x %d", ESPI_MASTER_INIT_PIN, ret);
+
+	printk("complete\n");
+	return ret;
+}
+
+int espi_handshake(void)
+{
+	int ret;
+
+	printk("eSPI test - Handshake with eSPI master...\n");
+
+	ret = wait_for_pin(gpio_devA, 1000, BOARD_PWRGD_PIN, 1);
+	if (ret) {
+		printk("Reset power good timeout!");
+		return ret;
+	}
+
+	printk("\t1st phase completed\n");
+	ret = wait_for_vwire(espi_dev, ESPI_VWIRE_SIGNAL_SUS_WARN,
+				60000, 1);
+	if (ret) {
+		printk("SUS_WARN Timeout!");
+		return ret;
+	}
+
+	printk("\t2nd phase completed\n");
+	ret = wait_for_vwire(espi_dev, ESPI_VWIRE_SIGNAL_SLP_S5, SLPS5_DLY, 1);
+	if (ret) {
+		printk("SLP_S5 Timeout!");
+		return ret;
+	}
+	ret = wait_for_vwire(espi_dev, ESPI_VWIRE_SIGNAL_SLP_S4, SLPS4_DLY, 1);
+	if (ret) {
+		printk("SLP_S4 Timeout!");
+		return ret;
+	}
+	ret = wait_for_vwire(espi_dev, ESPI_VWIRE_SIGNAL_SLP_S3, SLPS3_DLY, 1);
+	if (ret) {
+		printk("SLP_S3 Timeout!");
+		return ret;
+	}
+	printk("\t3rd phase completed\n");
+
+	ret = wait_for_pin(gpio_devB, 1000, SYSTEM_UP_PIN, 1);
+	if (ret) {
+		printk("SYSTEM_UP_PIN Timeout!");
+		return ret;
+	}
+
+	ret = gpio_pin_write(gpio_devC, ESPI_MASTER_RUN_PIN, 1);
+	if (ret) {
+		printk("Failed to indicate eSPI mster to run\n");
+		return ret;
+	}
+	ret = gpio_pin_write(gpio_devB, WAKE_SCI_PIN, 1);
+	if (ret) {
+		printk("Failed to indicate eSPI mster to run\n");
+		return ret;
+	}
+
+	ret = wait_for_vwire(espi_dev, ESPI_VWIRE_SIGNAL_PLTRST, 60000, 1);
+	if (ret) {
+		printk("PLT_RST Timeout!");
+		return ret;
+	}
+	printk("\t4th phase completed\n");
+	return 0;
+}
+
+void main(void)
+{
+	gpio_devA = device_get_binding(PORTA_LABEL);
+	if (gpio_devA) {
+		printk("%s FOUND!\n", PORTA_LABEL);
+	}
+	gpio_devB = device_get_binding(PORTB_LABEL);
+	if (gpio_devB) {
+		printk("%s FOUND!\n", PORTB_LABEL);
+	}
+	gpio_devC = device_get_binding(PORTC_LABEL);
```

Comment:
All these gpios are most probably necessary for your board, but will not be for another board (maybe less or maybe more will be required). You'll have to find a way to generalize this

Reply:
Put all board-specific in a Kconfig and board-specific prj.conf which showscase  eSPI minimal API functionality